addons/web/models/models.py

```diff
@@ -114,3 +121,150 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
         if r['type'] == 'qweb':
             r['arch'] = '<qweb/>'
         return r
+
+    @api.model
+    def get_search_panel_category(self, field_name):
+        """
+        Return possible category values, and their parent field.
+
+        :param field_name: the name of a many2one category field
+        :return: {
+            'parent_field': parent field on the comodel of field, or False
+            'values': array of dictionaries containing some info on the records
+                        available on the comodel of the field 'field_name'.
+                        The display name (and possibly parent_field) are fetched.
+        }
+        """
+        field = self._fields[field_name]
+        if field.type not in CATEGORY_SUPPORTED_TYPES:
+            raise UserError('Only types {} are supported for category (found type {})'.format(
```

Comment:
translate + name placeholders with %s() that transifex seems to better support

```suggestion
            raise UserError(_('Only types %(supported_types)s are supported for category (found type %(field_type)s)') % (
```